rb/poll.py

```diff
@@ -157,6 +160,43 @@ def poll(self, timeout=None):
         return rv
 
 
+def _is_closed_select(f):
+    rlist, wlist, _ = select.select([f], [f], [], 0.0)
+    if not rlist and not wlist:
+        return False
+    buf = array.array('i', [0])
+    fcntl.ioctl(f.fileno(), termios.FIONREAD, buf)
+    return buf[0] == 0
+
+
+def _is_closed_poll(f):
+    poll = select.poll()
+    poll.register(f.fileno(), select.POLLHUP)
+    for _, event in poll.poll(0.0):
+        if event == 'close':
+            return True
+    return False
+
+
+def _is_closed_kqueue(f):
+    kqueue = select.kqueue()
+    event = select.kevent(
+        f.fileno(), filter=select.KQ_FILTER_READ,
+        flags=select.KQ_EV_ADD | select.KQ_EV_ENABLE)
+    for event in kqueue.control([event], 128, 0.0):
+        if event.flags & select.KQ_EV_EOF:
+            return True
+    return False
+
+
+def is_closed(f):
+    if KQueuePoller.is_available:
```

Comment:
Why not just make these class methods on the polled class?

Then you can be like: `poll.is_closed`

Seems it'd be better to encapsulate the logic on the poller classes themselves rather than a bunch of random functions.